src/httpserver/http_request.hpp

```diff
@@ -301,6 +317,17 @@ class http_request {
          }
      }
 
+     /**
+      * Method used to set a file info into the map by key.
+      * @param upload_file_name The file name the user uploaded (identifying the map entry)
+      * @param file_system_file_name The path to the file in the file system (the name may not be the original name depending on configuration of the webserver)
+      * @param size The size in number of char of the file.
+     **/
+     void set_file(const char* upload_file_name, const char* file_system_file_name, size_t file_size) {
```

Comment:
Let's use strings references on method signatures as much as we can (I am fine with internal conversions, but I'd preserve the interface, if possible).